components/form/Select.js

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Label, ErrorMessage } from '.';
+import { useFormContext } from 'react-hook-form';
+
+export function SelectField({ className, containerClassName, label, labelClassName, name, ...options }) {
+  const {
+    register,
+    formState: { errors }
+  } = useFormContext();
+
+  const hasErrors = !!errors?.[name];
+
+  const errorClass = hasErrors ? 'focus:border-amcovad-danger' : 'focus:border-amcovad-primary-500';
+
+  const style = {
+    container: `relative my-1 `,
+    default: `${errorClass} relative flex flex-1 w-full mt-1 rounded-md py-2 px-2 bg-transparent text-amcovad-secondary-700 placeholder-amcovad-secondary-700 border border-amcovad-secondary-300 text-base focus:outline-none focus:ring-1 peer border-2 `
+  };
+
+  return (
+    <div className={`${style.container} ${containerClassName}`}>
+      <Label htmlFor={name} name={name} text={label} className={`!text-sm ${labelClassName}`} />
+      <select
+        className={`${style.default} ${className}`}
+        {...register(name)}
+        id={name}
+        name={name}
+        {...options}
```

Comment:
I am not sure why you added this but it looks wrong. In normal HTML, here is how the select tag is used:

```html
<select name="cars" id="cars">
  <option value="volvo">Volvo</option>
  <option value="saab">Saab</option>
  <option value="mercedes">Mercedes</option>
  <option value="audi">Audi</option>
</select>
```

From the above you can see that the `option` tags are inside the `select` tag. From the Select implementation, you generated the `options` tag as children:

```js
<SelectField label={label} name={name}>
        {defaultOption && <option value=""> {defaultOption}</option>}
        {options.map(({ title, value }, id) => {
         ...
            <option key={id} value={optionValue}>
              {optionTitle}
            </option>
          );
        })}
      </SelectField>
```

The best way to implement this is to destructure the `children` props from `SelectField` component and use them as generated options as children. 

```jsx
 <select
    className={`${style.default} ${className}`}
    {...register(name)}
    id={name}
    name={name}
>
   {children}
</select>
```